docs/designers-developers/developers/tutorials/javascript/js-build-setup.md

```diff
@@ -0,0 +1,228 @@
+
+## Quick Start
+
+For a quick start, you can use one of the examples from the [Gutenberg Examples repository](https://github.com/wordpress/gutenberg-examples/). Each one of the `-esnext` directories contain the necessary files for working with ESNext and JSX.
+
+## Setup
+
+Both Webpack and Babel are tools written in JavaScript and run using [Node.js](https://nodejs.org/). Node.js is a runtime environment for JavaScript outside of a browser. Simply put, node allows you to run JavaScript code on the command-line.
+
+First step is setting up node for your development environment. This is dependent on your operating system, there are node packages available for most package systems. If you are using one of the package systems, setup can be as straight-forward as:
+```
+macOS: brew install node
+Windows: choco install node
+Ubuntu: apt install node
+```
+
+Additionally, the [Node.js download page](https://nodejs.org/en/download/) includes installers and binaries.
+
+**Note:** The build tools and process occur on the command-line, so some basic familiarity using a terminal application is required. Some text editors have a terminal built-in which is fine to use, Visual Studio Code editor is one popular example.
+
+### Node Package Manager
+
+The Node Package Manager (npm) is a tool included with node. NPM allows you to install and manage JavaScript packages using the package.json definition file.
+
+First, create a directory to work in.
+```
+mkdir myguten-block
+```
+
+You create a new package.json using: `npm init`.  This will walk you through creating a file.
+
+```
+$ npm init
+
+package name: (myguten-block) myguten-block
+version: (1.0.0)
+description: Test block
+entry point: (index.js) block.js
+test command:
+git repository:
+keywords:
+author: mkaz
+license: (ISC) GPL-2.0-only
+About to write to /home/mkaz/src/wp/scratch/package.json:
+
+{
+  "name": "myguten-block",
+  "version": "1.0.0",
+  "description": "Test block",
+  "main": "block.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "mkaz",
+  "license": "GPL-2.0-only"
+}
+
+
+Is this OK? (yes) yes
+```
+
+## Webpack & Babel
+
+The next step is to install webpack. You can install packages using the npm command: `npm install`. If you pass the `--save` parameter, npm will write the package as a dependency in package.json file.
+
+Run: `npm install --save webpack`
+
+After installing, a `node_modules` directory is created with the webpack module and its dependencies.
+
+Also, if you look at package.json file it will include a new section:
+```json
+"dependencies": {
+	"webpack": "^4.29.0"
+}
+```
+
+Next, you configure webpack to process the `block.js` file and run babel to transform the JSX.
+
+Create the file `webpack.config.js`
+
+```js
+// sets mode webpack runs under
+const NODE_ENV = process.env.NODE_ENV || 'development';
```

Comment:
In #12685 @griffbrad started exploring how we could extract some bits of Webpack config to make it even simpler. We also discussed during [weekly core JS chat last week](https://make.wordpress.org/core/2019/01/23/javascript-chat-summary-january-22nd-2019/ ) the following:

> The discussion around build and start focused mostly on the question how to approach default configuration for plugins and themes. We’re considering to extract parts of the Webpack config as npm package to simplify setup for plugin developers. This would also allow to provide a default config for `@wordpress/scripts` and proposed scripts. To provide flexibility, it was raised that an eventual Webpack config package could have 3 types of exports:
> 
> A full webpack config with a simple preset entries map for simple starter plugins ( `front.js`, `admin.js`, `blocks.js` or something like that ) to provide a simple zero-config option.
> A function that takes an entries map and outputs the full webpack config with recommended values.
> Each part of the recommended config as their own keys.

We probably have to leave this manual as is, however we should keep in mind that we want to make it simpler for everyone as soon as possible :)

Reply:
:+1: